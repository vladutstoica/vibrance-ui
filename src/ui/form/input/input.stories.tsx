import type { Meta, StoryObj } from "@storybook/react";
import * as React from "react";
import { Label } from "../label/label";
import { Input } from "./input";

const meta: Meta<typeof Input> = {
  title: "Design System/Form/Input",
  component: Input,
  tags: ["autodocs"],
  argTypes: {
    type: {
      description: "The type of the input field",
      control: "select",
      options: [
        "text",
        "email",
        "password",
        "number",
        "search",
        "tel",
        "url",
        "file",
      ],
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "text" },
      },
    },
    disabled: {
      description: "Whether the input is disabled",
      control: "boolean",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    placeholder: {
      description: "Placeholder text for the input",
      control: "text",
    },
  },
  decorators: [
    (Story, context) => (
      <div className="grid w-full max-w-sm gap-1.5">
        <Label htmlFor={`input-${context.name}`}>{context.name}</Label>
        {Story({ args: { ...context.args, id: `input-${context.name}` } })}
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Input>;

export const Default: Story = {
  args: {
    placeholder: "Enter your text",
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
    placeholder: "Disabled input",
  },
};

export const WithValue: Story = {
  args: {
    defaultValue: "Hello World",
    "aria-label": "Text input with value",
  },
};

export const Email: Story = {
  args: {
    type: "email",
    placeholder: "Enter your email",
  },
};

export const Password: Story = {
  args: {
    type: "password",
    placeholder: "Enter your password",
  },
};

export const Search: Story = {
  args: {
    type: "search",
    placeholder: "Search...",
  },
};

export const File: Story = {
  args: {
    type: "file",
    accept: "image/*",
  },
};

export const WithError: Story = {
  render: (args) => (
    <div className="grid w-full max-w-sm gap-1.5">
      <Label htmlFor="input-error">Email</Label>
      <Input
        id="input-error"
        type="email"
        placeholder="Enter your email"
        className="border-red-500 focus-visible:ring-red-500"
        aria-describedby="email-error"
        {...args}
      />
      <p id="email-error" className="text-red-500 text-sm">
        Please enter a valid email address
      </p>
    </div>
  ),
};
