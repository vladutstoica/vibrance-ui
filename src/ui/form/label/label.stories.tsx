import type { Meta, StoryObj } from "@storybook/react";
import * as React from "react";
import { Checkbox } from "../checkbox/checkbox";
import { Input } from "../input/input";
import { Label } from "./label";

const meta: Meta<typeof Label> = {
  title: "Form/Label",
  component: Label,
  tags: ["autodocs"],
  argTypes: {
    htmlFor: {
      description: "The id of the form control this label is associated with",
      control: "text",
    },
  },
};

export default meta;
type Story = StoryObj<typeof Label>;

export const Default: Story = {
  args: {
    children: "Accept terms and conditions",
  },
};

export const WithInput: Story = {
  render: () => (
    <div className="grid w-full max-w-sm items-center gap-1.5">
      <Label htmlFor="email">Your email address</Label>
      <Input type="email" id="email" placeholder="Email" />
    </div>
  ),
};

export const WithCheckbox: Story = {
  render: () => (
    <div className="flex items-center space-x-2">
      <Checkbox id="terms" />
      <Label htmlFor="terms">Accept terms and conditions</Label>
    </div>
  ),
};

export const Required: Story = {
  render: () => (
    <div className="grid w-full max-w-sm items-center gap-1.5">
      <Label htmlFor="required-email">
        Email <span className="text-red-500">*</span>
      </Label>
      <Input type="email" id="required-email" placeholder="Email" required />
    </div>
  ),
};

export const WithDescription: Story = {
  render: () => (
    <div className="grid w-full max-w-sm gap-1.5">
      <Label htmlFor="picture">Picture</Label>
      <Input id="picture" type="file" />
      <p className="text-muted-foreground text-sm">
        Upload a picture to your profile.
      </p>
    </div>
  ),
};
