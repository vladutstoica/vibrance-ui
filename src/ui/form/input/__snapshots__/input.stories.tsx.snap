// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System/Form/Input Default smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="input-Default"
  >
    Default
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         placeholder="Enter your text"
         id="input-Default"
  >
</div>
`;

exports[`Design System/Form/Input Disabled smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="input-Disabled"
  >
    Disabled
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         disabled
         placeholder="Disabled input"
         id="input-Disabled"
  >
</div>
`;

exports[`Design System/Form/Input Email smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="input-Email"
  >
    Email
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         placeholder="Enter your email"
         id="input-Email"
         type="email"
  >
</div>
`;

exports[`Design System/Form/Input File smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="input-File"
  >
    File
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         accept="image/*"
         id="input-File"
         type="file"
  >
</div>
`;

exports[`Design System/Form/Input Password smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="input-Password"
  >
    Password
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         placeholder="Enter your password"
         id="input-Password"
         type="password"
  >
</div>
`;

exports[`Design System/Form/Input Search smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="input-Search"
  >
    Search
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         placeholder="Search..."
         id="input-Search"
         type="search"
  >
</div>
`;

exports[`Design System/Form/Input WithError smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="input-With Error"
  >
    With Error
  </label>
  <div class="grid w-full max-w-sm gap-1.5">
    <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
           for="input-error"
    >
      Email
    </label>
    <input class="flex h-10 w-full rounded-md border bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm border-destructive focus-visible:ring-destructive"
           id="input-With Error"
           aria-describedby="email-error"
           type="email"
    >
    <p id="email-error"
       class="text-red-700 text-sm dark:text-red-400"
       aria-live="polite"
    >
      Please enter a valid email address
    </p>
  </div>
</div>
`;

exports[`Design System/Form/Input WithValue smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="input-With Value"
  >
    With Value
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         aria-label="Text input with value"
         id="input-With Value"
         value="Hello World"
  >
</div>
`;
